File: CSharp/CSharp/CSharp/repository/ProbaDBRepository.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.SQLite;
using CSharp.domain;
using log4net;
namespace CSharp.repository
{
    public class ProbaDBRepository:IProbaRepository
    {
        private static readonly ILog log = LogManager.GetLogger("ProbaDBRepository");
        public ProbaDBRepository()
        {
            log.Info("Creating ProbaDBRepository");
        }
        public void Add(Proba elem)
        {
            log.InfoFormat("Add proba {0}",elem);
            var con = DBUtils.getConnection();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "insert into Probe(denumire,varstaMin,varstaMax) values (@denumire,@varstaMin,@varstaMax)";
                comm.Parameters.Add(new SQLiteParameter("@denumire", elem.Denumire));
                comm.Parameters.Add(new SQLiteParameter("@varstaMin", elem.VarstaMin));
                comm.Parameters.Add(new SQLiteParameter("@varstaMax", elem.VarstaMax));
                var result = comm.ExecuteNonQuery();
                if(result==0)
                    log.Error("Error Add proba");
                else
                {
                    log.Info("Successful adding");
                }
            }
        }

        public IEnumerable<Proba> FindAll()
        {
            IDbConnection con = DBUtils.getConnection();
            IList<Proba> probe = new List<Proba>();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from Probe";

                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        int idP = dataR.GetInt32(0);
                        String denumire = dataR.GetString(1);
                        int varstaMin = dataR.GetInt32(2);
                        int varstaMax = dataR.GetInt32(3);

                        Proba proba = new Proba(denumire, varstaMin, varstaMax);
                     
                        proba.Id = idP;
                        probe.Add(proba);
                    
                    }
                }
            }

            return probe;
        }

        public List<Proba> findAllByDenumire(string denumire)
        {
            IDbConnection con = DBUtils.getConnection();
            log.InfoFormat("Entering findOne with value {0}", denumire);
            List<Proba> probe = new List<Proba>();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from Probe where denumire=@denumire";
                IDbDataParameter paramDenumire = comm.CreateParameter();
                paramDenumire.ParameterName = "@denumire";
                paramDenumire.Value = denumire;
                comm.Parameters.Add(paramDenumire);

                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        int idP = dataR.GetInt32(0);
                        String denumire2 = dataR.GetString(1);
                        int varstaMin = dataR.GetInt32(2);
                        int varstaMax = dataR.GetInt32(3);

                        Proba proba = new Proba(denumire2, varstaMin, varstaMax);
                     
                        proba.Id = idP;
                        probe.Add(proba);
                    
                    }
                }
            }

            return probe;
            
        }

     

        public Proba FindOne(long id)
        {
            log.InfoFormat("Entering findOne with value {0}", id);
            IDbConnection con = DBUtils.getConnection();

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from Probe where id=@id";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        int idP = dataR.GetInt32(0);
                        String denumire = dataR.GetString(1);
                     
                        int varstaMin = dataR.GetInt32(2);
                        int varstaMax = dataR.GetInt32(3);

                        Proba proba = new Proba(denumire, varstaMin, varstaMax);
                       
                        proba.Id = idP;
                        
                        log.InfoFormat("Exiting findOne with value {0}", proba);
                        return proba;
                    }
                }
            }
            log.InfoFormat("Exiting findOne with value {0}", null);
            return null;
        }

        //de revazut
        public Proba findOneByDenumireVarsta(string denumire, int varstaMin, int varstaMax)
        {
            log.InfoFormat("Entering findOne with value {0}{1}{2}", denumire,varstaMin,varstaMax);
            IDbConnection con = DBUtils.getConnection();

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from Probe where denumire=@denumire and varstaMin=@varstaMin";
                IDbDataParameter paramId = comm.CreateParameter();
                IDbDataParameter paramId2 = comm.CreateParameter();
                IDbDataParameter paramId3 = comm.CreateParameter();
                paramId.ParameterName = "@denumire";
                paramId.Value = denumire;
                paramId2.ParameterName = "@varstaMin";
                paramId2.Value = varstaMin;
                paramId3.ParameterName = "@varstaMax";
                paramId3.Value = varstaMax;
                comm.Parameters.Add(paramId);
                comm.Parameters.Add(paramId2);
                comm.Parameters.Add(paramId3);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        int idP = dataR.GetInt32(0);
                        String denumire1 = dataR.GetString(1);
                     
                        int varstaMin1 = dataR.GetInt32(2);
                        int varstaMax1 = dataR.GetInt32(3);

                        Proba proba = new Proba(denumire1, varstaMin1, varstaMax1);
                       
                        proba.Id = idP;
                        
                        log.InfoFormat("Exiting findOne with value {0}", proba);
                        return proba;
                    }
                }
            }
            log.InfoFormat("Exiting findOne with value {0}{1}{2}", null);
            return null;
        }
    }
}

File: JAVApb5/ConcursPersistance/src/main/java/concurs/persistance/jdbc/ProbaDBRepository.java
package concurs.persistance.jdbc;

import concurs.domain.Participant;
import concurs.domain.Proba;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import concurs.persistance.ProbaRepository;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class ProbaDBRepository implements ProbaRepository {

    private JdbcUtils dbUtils;

    private static final Logger logger= LogManager.getLogger();

    public ProbaDBRepository(Properties props) {
        logger.info("Initializing ParticipantRepository with properties: {} ",props);
        dbUtils=new JdbcUtils(props);
    }


    @Override
    public void add(Proba elem) {
        logger.traceEntry("saving task{}",elem);
        Connection com=dbUtils.getConnection();
        logger.traceEntry("saving task{}",elem);
        Connection con = dbUtils.getConnection();
        try(PreparedStatement preStmt = con.prepareStatement("insert into \"Probe\"(denumire,\"varstaMin\",\"varstaMax\") values (?,?,?)")){
            preStmt.setString(1,elem.getDenumire());
            preStmt.setInt(2, elem.getVarstaMin());
            preStmt.setInt(3,elem.getVarstaMax());
            int result = preStmt.executeUpdate();
            logger.trace("Saved {} instances",result);

        } catch (SQLException e) {
            logger.error(e);
            System.err.print("Error DB "+e);
        }
        logger.traceExit();
    }

    @Override
    public Iterable<Proba> findAll() {

        logger.traceEntry();
        Connection conn = dbUtils.getConnection();

        System.out.println(conn);

        List<Proba> probe = new ArrayList<>();
        try (PreparedStatement preparedStatement = conn.prepareStatement("select * from \"Probe\"")) {
            try (ResultSet resultSet = preparedStatement.executeQuery()) {
                while (resultSet.next()) {
                    Long id = resultSet.getLong("id");
                    String denumire= resultSet.getString("denumire");
                    int varstaMin = resultSet.getInt("varstaMin");
                    int varstaMax = resultSet.getInt("varstaMax");
                    Proba proba=new Proba(denumire,varstaMin,varstaMax);
                    proba.setId(id);
                    probe.add(proba);
                }
            }
        } catch (SQLException e) {
            logger.error(e);
            System.err.print("Error DB "+e);
        }
        logger.traceExit(probe);
        //conn.close();
        return probe;

    }

    @Override
    public void update(Proba elem, Long aLong) {

        logger.traceEntry("updating request {} ",elem);
        Connection con=dbUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("update \"Probe\" set nrParticipanti=? where id=?")){

         //   preStmt.setInt(1,elem.getNrParticipanti());
            preStmt.setLong(2,elem.getId());
            int result=preStmt.executeUpdate();
            logger.trace("Updated {} instances",result);

        }catch (SQLException ex){
            logger.error(ex);
            System.err.println("Error DB "+ex);
        }
        logger.traceExit();
    }

    @Override
    public Iterable<Proba> findAllByDenumire(String denumire1) {

        logger.traceEntry();
        Connection conn = dbUtils.getConnection();

        System.out.println(conn);

        List<Proba> probe = new ArrayList<>();
        try (PreparedStatement preparedStatement = conn.prepareStatement("select * from \"Probe\"  where denumire='"+denumire1+ "'")) {

            try (ResultSet resultSet = preparedStatement.executeQuery()) {
                while (resultSet.next()) {
                    Long id = resultSet.getLong("id");
                    String denumire= resultSet.getString("denumire");
                    int varstaMin = resultSet.getInt("varstaMin");
                    int varstaMax = resultSet.getInt("varstaMax");
                    Proba proba=new Proba(denumire,varstaMin,varstaMax);
                    proba.setId(id);
                    probe.add(proba);
                }
            }
        } catch (SQLException e) {
            logger.error(e);
            System.err.print("Error DB "+e);
        }
        logger.traceExit(probe);
        //conn.close();
        return probe;

    }

    @Override
    public Proba findOne(Long aLong) {
        logger.traceEntry();
        Connection conn = dbUtils.getConnection();

        System.out.println(conn);

        Participant participant=null;
        Proba proba=null;
        try (PreparedStatement preparedStatement = conn.prepareStatement("select * from \"Probe\" where id=" +
                aLong )) {
            try (ResultSet resultSet = preparedStatement.executeQuery()) {
                while (resultSet.next()) {
                    Long id = resultSet.getLong("id");
                    String denumire = resultSet.getString("denumire");

                    int varstaMin = resultSet.getInt("varstaMin");
                    int varstaMax = resultSet.getInt("varstaMax");
                    proba =new Proba(denumire,varstaMin,varstaMax);

                    proba.setId(id);

                }
            }
        } catch (SQLException e) {
            logger.error(e);
            System.err.print("Error DB "+e);
        }
        logger.traceExit(proba);
        //conn.close();
        return proba;

    }




    @Override
    public Proba findOneByDenumireVarsta(String denumireDat, int varstaMinDat, int varstaMaxDat) {
        logger.traceEntry();
        Connection conn = dbUtils.getConnection();

        System.out.println(conn);


        Proba proba=null;
        try (PreparedStatement preparedStatement = conn.prepareStatement( "select * from \"Probe\" where denumire='"+denumireDat+"' and varstaMin='"+varstaMinDat+"'")) {
            try (ResultSet resultSet = preparedStatement.executeQuery()) {
                while (resultSet.next()) {
                    Long id = resultSet.getLong("id");
                    String denumire1 = resultSet.getString("denumire");


                    int varstaMin1 = resultSet.getInt("varstaMin");
                    int varstaMax1 = resultSet.getInt("varstaMax");


                    proba=new Proba(denumire1,varstaMin1,varstaMax1);
                    proba.setId(id);


                }
            }
        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }
        logger.traceExit(proba);
        //conn.close();
        return proba;
    }



}
